### DISCUSSION TITLE
onelight lualine background indistinguishable

### DISCUSSION THREAD

**Discussion by zhengpd** [olimorris/onedarkpro.nvim#244] [Q&A] [answered]
<img width="722" alt="image" src="https://github.com/user-attachments/assets/9be813b8-d6fa-4a32-aad7-d520f81da153">

For inactive windows, lualine background is the same as editor background. Is it supposed to be like that? Should we allow customizing the lualine bg color via `colors`?

**Comment by olimorris** [comment, marked as answer]
I don't use Lualine so I can't recall how it looks. Reading the [onelight file](https://github.com/olimorris/onedarkpro.nvim/blob/main/lua/lualine/themes/onelight.lua) I may have made a poor design choice that makes it difficult to apply your own customisations. I'd be open to a PR to fix this

**Comment by anjos** [comment]
I'm struggling with the same issue. [The issue comes from this line](https://github.com/olimorris/onedarkpro.nvim/blob/main/lua/lualine/themes/onedark.lua#L32), which appears on every `lua/lualine/themes/*.lua` variant:

```lua
local inactive_bg = config.options.highlight_inactive_windows and colors.color_column or colors.bg
                                                                                         ^^^^^^^^^
```

The choice of `colors.bg` is not very fortunate since it matches the default text editor background, making the split bar for inactive sessions nearly invisible (when you `:split` the screen), as you already saw. To improve contrast in the easiest way, this could be simply set via a oneliner patch to that file, so line 32 reads as such, replacing `colors.bg` by `colors.bg_statusline` instead:

```lua
local inactive_bg = config.options.highlight_inactive_windows and colors.color_column or colors.bg_statusline
                                                                                         ^^^^^^^^^^^^^^^^^^^^
```

`bg_statusline` is by default set to a darkened version of the background, so that would work all the time.  BTW, it is also the color that is used in the active lualine sections, so that would make it consistent at this level.  Slightly adjusting `bg_statusline` to a darker version of its current setting would improve contrast a lot, IMO, but that is only my own taste.

**Plan B**: If you dislike the theme setup for lualine altogether, you may still tweak the theme by noticing how lualine searches for the theme configuration when `theme = "auto"` is set on the lualine configuration - it searches for a file named `lua/lualine/themes/<colorscheme>.lua` on your runtime path (which I think by default includes `~/.config/nvim`). So, if you create a file named `~/.config/nvim/lualine/themes/onedark_vivid.lua` and set `colorscheme onedark_vivid`, then your file will be found first by lualine and loaded for that colorscheme configuration.  Copy [the original onedarkpro.nvim file](https://github.com/olimorris/onedarkpro.nvim/tree/main/lua/lualine/themes) to that location and override what you will.  For example, implement the above change and test it.
